bpf/bpf_host.c

```diff
@@ -1156,15 +1161,28 @@ do_netdev(struct __ctx_buff *ctx, __u16 proto, const bool from_host)
 
 		identity = resolve_srcid_ipv6(ctx, ip6, identity, &ipcache_srcid, from_host);
 		ctx_store_meta(ctx, CB_SRC_LABEL, identity);
-		if (from_host) {
+
 # if defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_MASQUERADE_IPV6)
+		if (from_host) {
 			/* If we don't rely on BPF-based masquerading, we need
 			 * to pass the srcid from ipcache to host firewall. See
 			 * comment in ipv6_host_policy_egress() for details.
 			 */
 			ctx_store_meta(ctx, CB_IPCACHE_SRC_LABEL, ipcache_srcid);
+		}
 # endif /* defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_MASQUERADE_IPV6) */
+
+#ifdef ENABLE_WIREGUARD
+		if (!from_host) {
+			next_proto = ip6->nexthdr;
+			l4_off = ETH_HLEN + ipv6_hdrlen(ctx, &next_proto);
```

Comment:
Note that `ipv6_hdrlen()` can return errors. Which we should probably tolerate in this case (since it can be arbitrary inbound traffic), but not continue in the WG check (since we wouldn't have extracted the actual L4 proto.).

Reply:
Correct, let me know in case we want to address it here, thanks for it!

Reply:
Resolving conversation, proposing fix in https://github.com/cilium/cilium/pull/35345.